src/plugins/intellisense/index.tsx

```diff
@@ -0,0 +1,512 @@
+import {
+  PartialFunctionCall,
+  TryFindMQIdentResult,
+  getController,
+  getMathquillIdentifierAtCursorPosition,
+  getPartialFunctionCall,
+} from "./latex-parsing";
+import { IntellisenseState } from "./state";
+import { View, addBracketsToIdent } from "./view";
+import { DCGView, MountedComponent, unmountFromNode } from "DCGView";
+import { MathQuillField, MathQuillView } from "components";
+import { Calc } from "globals/window";
+import { PluginController } from "plugins/PluginController";
+import { getMetadata } from "plugins/manage-metadata/manage";
+import { attach, propGetSet } from "utils/listenerHelpers";
+
+export type BoundIdentifier =
+  | {
+      exprId: string;
+      variableName: string;
+      type:
+        | "variable"
+        | "function-param"
+        | "listcomp-param"
+        | "substitution"
+        | "derivative"
+        | "repeated-operator";
+      id: number;
+    }
+  | BoundIdentifierFunction;
+
+export interface BoundIdentifierFunction {
+  exprId: string;
+  variableName: string;
+  type: "function";
+  id: number;
+  params: string[];
+}
+
+export function getMQCursorPosition(focusedMQ: MathQuillField) {
+  return getController(
+    focusedMQ
+  ).cursor?.cursorElement?.getBoundingClientRect();
+}
+
+const intellisenseMountPoint = document.createElement("div");
+document.body.appendChild(intellisenseMountPoint);
```

Comment:
Code that runs at top-level is a warning flag for me. This seems pretty harmless (since the div is typically invisible), but this'll need to change if you want to mount to the inside of the scrolled inner element.

Reply:
By "top-level" are you referring to top-level in the TS/JS or in the HTML?

Reply:
I'm referring to code running when a file is imported. In this case, `intellisenseMountPoint` is defined once throughout the execution of the program, which is fine since DesModder runs only once per page. But it gets defined in some loading order that's based on when files are imported. This risks some problems if the side effects (from loading the file) are significant. Here there's not really a problem though.

Reply:
I've moved it to the Intellisense class. It also now creates/destroys `intellisenseMountPoint` when the plugin is enabled/disabled.